ccmlib/scylla_repository.py

```diff
@@ -181,6 +184,21 @@ def read_build_manifest(url):
     return yaml.safe_load(res.content)
 
 
+def normelize_scylla_version(version):
+    """
+    take 2020.2.rc3 or 2020.2.0.rc4 or 2020.2.0~rc5
+    and formolize them in to a semver base version 2020.2.0~rc5
+    """
+    version = version.replace('-', '~').replace('.rc', '~rc')
+    version = version if version.count('0~rc') else version.replace('~rc', '.0~rc')
+
+    return version
+
+def extract_major_version(version):
+    major_version = version if version.count('.') == 1 else '.'.join(n for n in version.split('.')[:2])
+    major_version = major_version.split('~')[0]
```

Comment:
what about `'-'`?

Reply:
it by now normalized to be only ~

Reply:
what if we backport this patch to older branches?

Reply:
It should work as well on older branches.

keep in mind we on only use rcs manually.

In CI/Jenkins we only run from relocate getting picked by scylla-pkg

we use the release option only on upgrade tests and only with formal releases

Reply:
Anyhow I would refactor this code a bit more

So it can have integration tests (without downloading the whole relocatable file)

Reply:
actually had to treat old versions a bit differently, but not here